tensorflow_addons/metrics/cohens_kappa.py

```diff
@@ -171,6 +171,13 @@ def _update_confusion_matrix(self, y_true, y_pred, sample_weight):
         y_true = tf.squeeze(y_true)
         y_pred = tf.squeeze(y_pred)
 
+        y_true = tf.cond(
+            tf.rank(y_true) == 0, lambda: tf.expand_dims(y_true, 0), lambda: y_true
+        )
+        y_pred = tf.cond(
+            tf.rank(y_pred) == 0, lambda: tf.expand_dims(y_pred, 0), lambda: y_pred
+        )
```

Comment:
Please use
```python
if tf.rank(y_true) == 0:
    y_true = tf.expand_dims(y_true, 0)
if tf.rank(y_pred) == 0:
    y_pred = tf.expand_dims(y_pred, 0)
```
instead.

Reply:
Actually this was my first version, but it gave me the error:

`OperatorNotAllowedInGraphError: using a tf.Tensor as a Python bool is not allowed: AutoGraph did not convert this function. Try decorating it directly with @tf.function.`

That's why I changed to `tf.cond` implementation, which is has been prohibited.

My last commit address both requirements, using a `@tf.function`:
```python
    @tf.function
    def _safe_squeeze(self, y):
        y = tf.squeeze(y)

        # Check for scalar result
        if tf.rank(y) == 0:
            y = tf.expand_dims(y, 0)

        return y

    def _update_confusion_matrix(self, y_true, y_pred, sample_weight):
        y_true = self._safe_squeeze(y_true)
        y_pred = self._safe_squeeze(y_pred)

        # ...
```